controllers/runner_controller.go

```diff
@@ -832,9 +832,9 @@ func newRunnerPodWithContainerMode(containerMode string, template corev1.Pod, ru
 
 	if containerMode == "kubernetes" {
 		if dockerdContainer != nil {
-			template.Spec.Containers = append(template.Spec.Containers[:dockerdContainerIndex], template.Spec.Containers[dockerdContainerIndex+1:]...)
+			template.Spec.Containers = append(template.Spec.Containers[dockerdContainerIndex+1:], template.Spec.Containers[:dockerdContainerIndex]...)
```

Comment:
```suggestion
			template.Spec.Containers = append(template.Spec.Containers[:dockerdContainerIndex], template.Spec.Containers[dockerdContainerIndex+1:]...)
```

My fault- this was fine. Reversing the append args result in unnecessarily reversing the order of slice elements. Let me revert it.